File: problems/958_check_completeness_of_a_binary_tree/solution.py
from typing import Optional
from shared.python.data_structures import TreeNode


class Solution:
    def isCompleteTree(self, root: Optional[TreeNode]) -> bool:
        from collections import deque
        queue = deque([root])
        found_null = False
        while queue:
            node = queue.popleft()
            if not node:
                found_null = True
            else:
                if found_null:
                    return False
                queue.append(node.left)
                queue.append(node.right)
        return True

File: problems/958_check_completeness_of_a_binary_tree/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    bool isCompleteTree(TreeNode* root) {
        queue<TreeNode*> q;
        q.push(root);
        bool foundNull = false;
        while (!q.empty()) {
            TreeNode* node = q.front(); q.pop();
            if (!node) foundNull = true;
            else {
                if (foundNull) return false;
                q.push(node->left);
                q.push(node->right);
            }
        }
        return true;
    }
};
